builders/tools/sidecar.md

```diff
@@ -143,36 +200,98 @@ To obtain the EVM sender address, recipient address, and EVM hash, check the `ev
+    | `Sender Address`       | extrinsics.{extrinsic number}.events.{event number}.data.0 |
+    | `Recipient Address`    | extrinsics.{extrinsic number}.events.{event number}.data.1 |
+    | `EVM Hash`             | extrinsics.{extrinsic number}.events.{event number}.data.2 |
+    | `EVM Execution Status` | extrinsics.{extrinsic number}.events.{event number}.data.3 |
+
 
 !!! note
-    EVM transaction nonce and signature fields are under `extrinsics.{extrinsic number}.args.transaction`, whereas the `nonce` and `signature` fields under `extrinsics.{extrinsic number}` are the Substrate transaction nonce and signature, which are set to `null` for EVM transactions.
+    EVM transaction nonce and signature fields are under `extrinsics.{extrinsic number}.args.transaction.{transactiontype}`, whereas the `nonce` and `signature` fields under `extrinsics.{extrinsic number}` are the Substrate transaction nonce and signature, which are set to `null` for EVM transactions.
 
 
-### Computing Gas Spent {: #computing-gas-used } 
+### Computing Gas Used {: #computing-gas-used } 
 
 To calculate the gas spent or used during EVM execution of the transaction, the following formula can be used: 
 
-```
-GasPrice * Tx Weight / {{ networks.moonbase.tx_weight_to_gas_ratio }}
-```
 
-where GasPrice can be read from the block according to the above table, and Tx Weight can be retrieved under the event of the relevant extrinsic where the `method` field is set to: 
+=== "EIP1559"
+    ```
+    Gas Used =（Base Fee + MaxPriorityFeePerGas) * Tx Weight / {{ networks.moonbase.tx_weight_to_gas_ratio }}
+    ```
+=== "Legacy"
+    ```
+    Gas Used = GasPrice * Tx Weight / {{ networks.moonbase.tx_weight_to_gas_ratio }}
+    ```
+=== "EIP2930"
+    ```
+    Gas Used = GasPrice * Tx Weight / {{ networks.moonbase.tx_weight_to_gas_ratio }}
+    ```
+
+The values of `GasPrice` and `MaxPriorityFeePerGas` for the applicable transaction types can be read from the block according to the above table. 
+
+The `Base Fee` for `EIP1559` type transactions is current static on Moonbeam networks and has the following assigned value:
+
+=== "Moonbeam"
+    |      Variable      |                                     Value                                      |
+    |:------------------:|:------------------------------------------------------------------------------:|
+    | Base Fee | 100 Gwei |
+
+=== "Moonriver"
+    |      Variable      |                                     Value                                      |
+    |:------------------:|:------------------------------------------------------------------------------:|
+    | Base Fee | 1 Gwei |
+
+=== "Moonbase Alpha"
+    |      Variable      |                                     Value                                      |
+    |:------------------:|:------------------------------------------------------------------------------:|
+    | Base Fee | 1 Gwei |
```

Comment:
is Base Fee a variable? should there be a space in between it?

Reply:
Base Fee is what it is, it's not an actual field in the JSON object, because we currently actually don't have a field for this in the block.